assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -520,7 +529,23 @@ const ExternalActivity: React.FC<PartComponentProps<CapiIframeModel>> = (props)
     const filterVars = createCapiObjectFromStateVars(simLife.currentState, simLife.domain);
     if (filterVars && Object.keys(filterVars)?.length !== 0) {
       handleIFrameSpecificProperties(simLife.currentState, simLife.domain);
-      sendFormedResponse(simLife.handshake, {}, JanusCAPIRequestTypes.VALUE_CHANGE, filterVars);
+      writeCapiLog('SENDING SIM CONFIG DATA !!!!', 3, {
+        simLife,
+        configData: filterVars,
+      });
+      Object.keys(filterVars).forEach((variable: any) => {
+        const formatted: Record<string, unknown> = {};
+        const simVariable = filterVars[variable];
+        const typeOfValue = typeof simVariable.value;
+        //Now we know that configData variable can contain a expression in value field so we need to parse that before sending it to SIM
+        if (typeOfValue === 'string' && simVariable.value.indexOf('{') !== -1) {
```

Comment:
I don't think that you *need* to check for the curly brace here, the templatizeText will handle that check

Reply:
I was only checking to avoid processing each text if it does not have expression. I will remove that.